src/core/crypto/crypto_platform.cpp

```diff
@@ -51,9 +54,22 @@
 using namespace ot;
 using namespace Crypto;
 
+#if OPENTHREAD_CONFIG_CRYPTO_LIB != OPENTHREAD_CONFIG_CRYPTO_LIB_PSA
+
 //---------------------------------------------------------------------------------------------------------------------
 // Default/weak implementation of crypto platform APIs
 
+#if (!defined(MBEDTLS_NO_DEFAULT_ENTROPY_SOURCES) && \
+     (!defined(MBEDTLS_NO_PLATFORM_ENTROPY) || defined(MBEDTLS_HAVEGE_C) || defined(MBEDTLS_ENTROPY_HARDWARE_ALT)))
+#define OT_MBEDTLS_STRONG_DEFAULT_ENTROPY_PRESENT
+#endif
+
+#if !OPENTHREAD_RADIO
+static mbedtls_ctr_drbg_context sCtrDrbgContext;
```

Comment:
Wouldn't it be more future-proof if the context was passed via `otCryptoContext` like in other crypto functions? For example, what if a single application runs multiple OT instances?

Reply:
Or you did so, because `Random::Crypto::MbedTlsContextGet()` is global anyway?

Reply:
Yes, currently there are no multi-instances of RNGs algorithms in OT. Let me maybe push this change and explicitly ask OT community whether they envision a use-case for the future, and if so, i will add `otCryptoContext` param to `otPlatCryptoRandom*` functions.